src/Umbraco.Core/Persistence/Repositories/Implement/LanguageRepository.cs

```diff
@@ -250,7 +250,6 @@ public ILanguage GetByIsoCode(string isoCode)
             lock (_codeIdMap)
             {
                 if (_codeIdMap.TryGetValue(isoCode, out var id)) return id;
-                if (isoCode.Contains('-') && _codeIdMap.TryGetValue(isoCode.Split('-').First(), out var invariantId)) return invariantId;
```

Comment:
@kjac are you sure about this change?? IIRC there's a reason for this check and the unit test that has been removed. Because we have odd language files that are only 2 chars but we have one that is properly named `en-us`

Reply:
Ah, maybe this was a change you made at some point - i guess my only concern is that there is some odd code in here to deal with the poorly named localization files that are both 2char and the normal iso codes

Reply:
@Shazwazza yah this PR reverts the changes made by yours truly in #4519 and re-implements the same feature in another way, as #4519 introduced issues for uSync (and was likely to do so for others in the future as well). 

All of this only concerns the dictionary, not the language files. Here's what I'm trying to achieve:

You're able to create region-less languages for frontend rendering (e.g. "da") and thus you have a region-less dictionary items. However you can't have a region-less user locale - it's always region specific (e.g. "da-DK"). So these changes are trying to bridge that gap, allowing for localizing the backoffice for the region specific user locales using region-less dictionary items.